src/main/java/be/seeseemelk/mockbukkit/entity/LivingEntityMock.java

```diff
@@ -78,44 +80,30 @@ public boolean isDead()
 	@Override
 	public void setHealth(double health)
 	{
-		if (health <= 0)
-		{
-			this.health = 0;
+		if (health <= 0) this.kill();
+		else this.health = Math.min(health, getMaxHealth());
+	}
 
-			if (this instanceof Player)
-			{
-				Player player = (Player) this;
-				List<ItemStack> drops = new ArrayList<>();
-				drops.addAll(Arrays.asList(player.getInventory().getContents()));
-				PlayerDeathEvent event = new PlayerDeathEvent(player, drops, 0, getName() + " got killed");
-				Bukkit.getPluginManager().callEvent(event);
-
-				// Terminate any InventoryView and the cursor item
-				player.closeInventory();
-
-				// Clear the Inventory if keep-inventory is not enabled
-				if (!getWorld().getGameRuleValue(GameRule.KEEP_INVENTORY).booleanValue())
-				{
-					player.getInventory().clear();
-					// Should someone try to provoke a RespawnEvent, they will now find the Inventory to be empty
-				}
-
-				player.setLevel(0);
-				player.setExp(0);
-				player.setFoodLevel(0);
-			}
-			else
-			{
-				EntityDeathEvent event = new EntityDeathEvent(this, new ArrayList<>(), 0);
-				Bukkit.getPluginManager().callEvent(event);
-			}
+	/**
+	 * Simulates a kill by setting the killer of this entity and propagating the necessary event
+	 * @param player the player who killed this entity
+	 */
+	public void simulateKilledBy(@Nullable Player player)
+	{
+		this.setKiller(player);
+		this.kill();
+	}
 
-			alive = false;
-		}
-		else
-		{
-			this.health = Math.min(health, getMaxHealth());
-		}
+	/*
+	 * Kills this entity and propagates the necessary event
+	 */
+	public void kill()
+	{
+		this.health = 0;
+		if (!this.alive) return;
+		EntityDeathEvent event = new EntityDeathEvent(this, new ArrayList<>(), 0);
```

Comment:
(may be a bit out of scope) ideally we would allow drops and exp here.
We could just add a method reference here for methods which just default to an empty list and 0 xp.
Then let programmers implement them if they desire so.

The according entities (Item and ExperienceOrb) would need to be created too if those are provided.
Again, probably out of scope for this but just something I wanted to note for the future.

```suggestion
		EntityDeathEvent event = new EntityDeathEvent(this, getDrops(), getDroppedExp());
```

Reply:
My Bad, I didn't think about it. I could simple compute drops from inventory (and getting them from different methods to allow overrides)